app/FinTsFactory.php

```diff
@@ -13,8 +14,8 @@
 class FinTsFactory
 {
     static function create_from_session(Session $session)
-    {
-        foreach(['bank_url','bank_code','bank_username','bank_password','bank_2fa'] as $required_session_value){
+    {   
+        foreach(['bank_url','bank_code','bank_username','bank_2fa'] as $required_session_value){
```

Comment:
Instead of simply removing the "password is set" sanity check here, could you replace it with an equivalent check using the new system?